pallets/fiat_filter/src/lib.rs

```diff
@@ -0,0 +1,225 @@
+        + IsSubType<blob::Call<Self>>
+        + IsSubType<revoke::Call<Self>>
+        + IsSubType<attest::Call<Self>>;
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// Call is not among the core_mods ones that should go through fiat_filter
+        UnexpectedCall,
+    }
+}
+
+// The module's callable functions (Dispatchable functions)
+// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+decl_module! {
+    pub struct Module<T: Config> for enum Call
+    where
+        origin: T::Origin
+    {
+        // Errors must be initialized if they are used by the pallet.
+        type Error = Error<T>;
+
+        /// Execute a Call. Must be a core_mods call (DID, Blob, Anchor, Revoke/Unrevoke, Attest)
+        #[weight = 10_000]
+        pub fn execute_call(origin, call: Box<<T as Config>::Call>) -> DispatchResultWithPostInfo {
+            Ok(Self::execute_call_(origin, &call)?)
+        }
+    }
+}
+
+type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as system::Config>::AccountId>>::Balance;
+
+type AmountUsd = u32;
+
+pub mod fiat_rate {
+    // all prices given in nUSD (billionth USD)
+    // This unit is chosen to avoid multiplications later in compute_call_fee_dock_()
+    pub const PRICE_DID_OP: u32 = 100_000; // 100_000/1B or 0.0001 USD
+    pub const PRICE_ANCHOR_OP_PER_BYTE: u32 = 2000;
+    pub const PRICE_BLOB_OP_PER_BYTE: u32 = 2000;
+    pub const PRICE_REVOKE_REGISTRY_OP: u32 = 100_000;
+    pub const PRICE_REVOKE_OP_CONST_FACTOR: u32 = 50_000;
+    pub const PRICE_REVOKE_PER_REVOCATION: u32 = 20_000;
+    pub const PRICE_ATTEST_PER_IRI_BYTE: u32 = 2000;
+
+    // minimum price, in case the price fetched from optimized_get_dock_usd_price is zero or an error
+    // expressed in USD_1000th/DOCK (as u32) (== USD/1000DOCK) just like the aactual result from optimized_get_dock_usd_price
+    pub const MIN_RATE_DOCK_USD: u32 = 1;
+}
+
+// private helper functions
+impl<T: Config> Module<T> {
+    /// Get fee in fiat unit for a given Call
+    /// Result expressed in nUSD (billionth USD)
+    fn get_call_fee_fiat_(call: &<T as Config>::Call) -> Result<AmountUsd, DispatchError> {
+        // TODO make sure we discuss and decide the actual pricing for each call type
+        match call.is_sub_type() {
+            Some(did::Call::new(_did, _detail)) => return Ok(fiat_rate::PRICE_DID_OP), // 50/1B or 0.00000005 USD
+            Some(did::Call::update_key(_key_update, _sig)) => return Ok(fiat_rate::PRICE_DID_OP),
+            Some(did::Call::remove(_to_remove, _sig)) => return Ok(fiat_rate::PRICE_DID_OP),
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(anchor::Call::deploy(bytes)) => {
+                return Ok(fiat_rate::PRICE_ANCHOR_OP_PER_BYTE.saturating_mul(bytes.len() as u32))
+            }
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(blob::Call::new(blob, _sig)) => {
```

Comment:
Regarding blob and attestation, there would be a base price and then for each 100 bytes, there would be extra cost so the final price is `base_price + ceil(size/100)*price_per_100`